treeherder/webapp/api/infra_compare.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+import time
+
+from rest_framework import generics
+from treeherder.model import models
+
+from .infra_serializers import InfraCompareSerializer, InfraCompareQuerySerializers
+
+from rest_framework.response import Response
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+
+class InfraCompareView(generics.ListAPIView):
+    serializer_class = InfraCompareSerializer
+    queryset = None
+
+    def list(self, request):
+        query_params = InfraCompareQuerySerializers(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
+
+        startday = query_params.validated_data['startday']
+        endday = query_params.validated_data['endday']
+        project = query_params.validated_data['project']
+        revision = query_params.validated_data['revision']
+        interval = query_params.validated_data['interval']
+        repository = models.Repository.objects.get(name=project)
+        push = models.Push.objects.filter(repository=repository).first()
+        job = models.Job.objects.filter(push=push)
+        if revision:
+            job = job.filter(push__revision=revision)
+        elif interval and not startday and not endday:
+            job = job.filter(
+                start_time__gt=datetime.datetime.utcfromtimestamp(int(time.time() - int(interval)))
```

Comment:
is time.time() in seconds or usecs and interval is in what format?  I assume interval will be hours

Reply:
time.time() is in seconds
Interval will be coming from frontend and I will have to make sure it is in seconds

Reply:
1) can you make a comment in here to indicate that
2) I look forward to seeing this converted and confirmed to be the right range.